### PR TITLE
Multi image support

### PR DESCRIPTION
**Pull request by wild-endeavor**
Example of Multi-Image support
https://github.com/lyft/flytesnacks/pull/44

### CODE REVIEW

File: flytekit/annotated/context_manager.py [hunk 1]
```diff
@@ -52,8 +113,8 @@ def version(self) -> str:
         return self._version
 
     @property
-    def image(self) -> str:
-        return self._image
+    def image(self) -> ImageConfig:
```
Comment (wild-endeavor) [inline flytekit/annotated/context_manager.py:116]: can we rename this to image_config?

File: flytekit/annotated/task.py [hunk 2]
```diff
@@ -764,6 +826,7 @@ def task(
     interruptible: bool = False,
     deprecated: str = "",
     timeout: Union[_datetime.timedelta, int] = 0,
+    container_image: str = None,
```
Comment (wild-endeavor) [inline flytekit/annotated/task.py:829]:
```suggestion
    container_image: Optional[str] = None,
```

Comment (wild-endeavor) [reply flytekit/annotated/task.py:829]: and in the @property too

File: tests/flytekit/unit/annotated/test_serialization.py [hunk 3]
```diff
@@ -102,11 +118,47 @@ def my_wf(a: int) -> str:
...
     )
     with ctx.current_context().new_registration_settings(registration_settings=registration_settings):
         wf = my_wf.get_registerable_entity()
         assert wf is not None
         assert len(wf.nodes) == 2
         assert wf.nodes[1].branch_node is not None
+
+
+def test_serialization_images():
+    @task(container_image="{{.image.xyz.fqn}}:{{.image.default.version}}")
```
Comment (wild-endeavor) [inline tests/flytekit/unit/annotated/test_serialization.py:137]: can we add a couple tests with hardcoded strings as well?  like without the {{}}

Comment (kumare3) [reply tests/flytekit/unit/annotated/test_serialization.py:137]: done!

File: flytekit/annotated/task.py (hunk 2 above)
Comment (kumare3) [reply flytekit/annotated/task.py:829]: Done

**Comment by kumare3** [comment]
the check was successful, but it does not update here

**Pull request merged**